src/compiler/scala/tools/nsc/ast/TreeGen.scala

```diff
@@ -308,8 +308,17 @@ abstract class TreeGen extends scala.reflect.internal.TreeGen with TreeDSL {
     val useMethodParams = new TreeSymSubstituter(fun.vparams.map(_.symbol), methParamSyms)
     // we're now owned by the method that holds the body, and not the function
     val moveToMethod = new ChangeOwnerTraverser(fun.symbol, methSym)
+    def substThisForModule(tree: Tree) = {
+      // Rewrite This(enclModuleClass) to Ident(enclModuleClass) to avoid unnecessary capture of the module
+      // class, which might hamper serializability.
+      //
+      // Analagous to this special case in ExplicitOuter: https://github.com/scala/scala/blob/d2d33ddf8c/src/compiler/scala/tools/nsc/transform/ExplicitOuter.scala#L410-L412
+      // that understands that such references shouldn't give rise to outer params.
+      val enclosingStaticModules = owner.enclClassChain.filter(x => !x.hasPackageFlag && x.isModuleClass && x.isStatic)
+      enclosingStaticModules.foldLeft(tree)((tree, moduleClass) => tree.substituteThis(moduleClass, gen.mkAttributedIdent(moduleClass.sourceModule)) )
```

Comment:
This could be fused into a single transform with some effort to generalize `ThisSubstituter` and `SubstThisMap`. That would reduce the overhead for lambdas owned by a chain of static modules.